samples/android/BeaconServiceDemoApp/app/src/main/java/com/google/sample/beaconservice/MainActivityFragment.java

```diff
@@ -245,6 +245,19 @@ private void createScanner() {
   }
 
   @Override
+  public void onResume() {
+    super.onResume();
+
+    // There could be multiple instances when we need to handle 'UserRecoverableAuthExceptions'.
+    // (i.e to updated GMS, to re-login, for permission to use the proximity beacon api)
+    // So run this check every time another activity has finished running.
+    String accountName;
+    if ((accountName = getActivity().getSharedPreferences(Constants.PREFS_NAME, 0).getString("accountName", "")) != "") {
```

Comment:
i guess we all typo the != string equality java gotcha once a year. i'll fix it up tho.  :-)